cmd/src/search.go

```diff
@@ -54,13 +54,29 @@ Other tips:
 		explainJSONFlag = flagSet.Bool("explain-json", false, "Explain the JSON output schema and exit.")
 		apiFlags        = api.NewFlags(flagSet)
 		lessFlag        = flagSet.Bool("less", true, "Pipe output to 'less -R' (only if stdout is terminal, and not json flag)")
+		streamFlag      = flagSet.Bool("stream", false, "Consume results as stream.")
+
+		// Streaming.
+		_ = flagSet.Int("display", -1, "Limit the number of results shown. Only supported for streaming.")
 	)
 
 	handler := func(args []string) error {
 		if err := flagSet.Parse(args); err != nil {
 			return err
 		}
 
+		if *streamFlag {
+			// Remove -stream from args.
+			argsWOStream := make([]string, 0, len(args)-1)
+			for _, a := range args {
+				if a == "-stream" {
+					continue
+				}
+				argsWOStream = append(argsWOStream, a)
+			}
+			return streamHandler(argsWOStream)
+		}
+
```

Comment:
If you define `-stream` as a noop on the other flagset that you then use in `streamHandler` you wouldn't have to do this here, right?

Reply:
yes, but then we mess with the auto-documentation. Suddenly the "streaming search" has a "-stream" flag which is odd. I traded odd code against odd documentation.

Reply:
Why not create a separate `stream-search` command?

Reply:
It is based on a discussion we had around exhaustive search which is documented in [RFC 288](https://docs.google.com/document/d/1dk309wEXA34b-LF66SBOzcNBJ6jZbWXT_eeBL9TZ2Ro/edit). At some point we want phase out -stream and replace search with streaming search. That is part of the reason why I wrote it as handler.

Reply:
This is kind of weird though:

```shellsession
$ ./src search -stream -get-curl foo
flag provided but not defined: -get-curl
Usage of streaming search:
  -display int
    	Limit the number of results that are displayed. Note that the statistics continue to report all results. (default -1)
  -insecure-skip-verify
    	Skip validation of TLS certificates against trusted chains
  -trace
    	Log the trace ID for requests. See https://docs.sourcegraph.com/admin/observability/tracing
```

Given we intend on replacing search with stream search in the future, can't we make the stream search stuff not use its own flagset. I would think an approach like

```go
streaming.Search(streaming.Opts{
  Query: flagSet.Args(),
  Trace: *trace,
  Display: *display,
  InsecureSkipVerify: *insecureSkipVerify,
})
```

would make sense. Or even you pass in the http client you use that hides the trace/insecure stuff.

That way you can also move the large amounts of business logic out of `./cmd/src` into `./internal/streaming`

Reply:
I actually like this behavior, but since both reviewers commented on this I may be on the track here. I refactored it to be a simple function call instead of a handler and moved parts of the logic to internal/streaming.